### ISSUE TITLE
The approx bleu score on the valid set  of en_vi translation  is 65,  but the output of the t2t-decoder is empty.

### ISSUE THREAD

**Issue by zhaoguangxiang** [tensorflow/tensor2tensor#1309]
### Description

I am training my model(a variant of transformer) on the en vi translation.  Transformer behaves normally on the valid set, and our model **behaves much better** than the transformer on the valid set .However,the decoder output of our model  is empty and the decode speed is much slower.

### Environment information
tensor2tensor 1.11
tensorflow-gpu 1.12.0
### dataset description
the dataset is en vi tranlation
the valid set is envi 2012 and the test set is envi 2013

train:
t2t-trainer --data_dir=t2t_data --problem=translate_envi_iwslt32k --model=transformer --hparams_set=transformer_base  --output_dir=t2t_output --hparams='XXX'

decode:
t2t-decoder --data_dir=t2t_data --problem=translate_envi_iwslt32k --model=transformer --decode_hparams="beam_size=4,alpha=0.6"  --decode_from_file=t2t_datagen/tst2013.en --decode_to_file=system.output  --hparams_set=transformer_base --output_dir=t2t_output --hparams='XXX'

**the normal transformer :**
**performace on the valid set:**
INFO:tensorflow:Saving dict for global step 63000: global_step = 63000, loss = 2.741762, metrics-translate_envi_iwslt32k/targets/accuracy = 0.56709224, metrics-translate_envi_iwslt32k/targets/accuracy_per_sequence = 0.007083065, metrics-translate_envi_iwslt32k/targets/accuracy_top5 = 0.77644753, metrics-translate_envi_iwslt32k/targets/approx_bleu_score = 0.26265332, metrics-translate_envi_iwslt32k/targets/neg_log_perplexity = -2.8089416, metrics-translate_envi_iwslt32k/targets/rouge_2_fscore = 0.3386241, metrics-translate_envi_iwslt32k/targets/rouge_L_fscore = 0.5353232
**decode output on the test set:**
  Output is normal,and the t2t-bleu is 30.3

**our model :**
**performace on the valid set**
_INFO:tensorflow:Saving dict for global step 60000: global_step = 60000, loss = 0.745806, metrics-translate_envi_iwslt32k/targets/accuracy = 0.898024, metrics-translate_envi_iwslt32k/targets/accuracy_per_sequence = 0.555699, metrics-translate_envi_iwslt32k/targets/accuracy_top5 = 0.970387, metrics-translate_envi_iwslt32k/targets/approx_bleu_score = 0.671908, metrics-translate_envi_iwslt32k/targets/neg_log_perplexity = -0.528144, metrics-translate_envi_iwslt32k/targets/rouge_2_fscore = 0.729822, metrics-translate_envi_iwslt32k/targets/rouge_L_fscore = 0.786252_

All metrics are extremly high.

**decode output on the test set**
INFO:tensorflow:Inference results INPUT: Absolutely .
INFO:tensorflow:Inference results OUTPUT:
INFO:tensorflow:Inference results INPUT: Extraordinary .
INFO:tensorflow:Inference results OUTPUT:
INFO:tensorflow:Elapsed Time: 5282.34019

and the t2t-bleu is 0
